stroyprombeton/models.py

```diff
@@ -47,6 +47,30 @@ def get_absolute_url(self):
         """Return url for model."""
         return reverse('category', args=(self.id,))
 
+    def get_series(self) -> models.QuerySet:
+        return (
+            Series.objects
+            .prefetch_related('options')
+            .filter(
+                options__in=models.Subquery(
+                    Option.objects
+                    .select_related('product')
+                    .select_related('product__category')
+                    .filter(
+                        product__category__in=(
+                            Category.objects
+                            .select_related('parent')
+                            .prefetch_related('children')
+                            .filter(id=self.id)
+                            .get_descendants(include_self=True)
```

Comment:
```suggestion
                            self.get_descendants(include_self=True)
```
https://django-mptt.readthedocs.io/en/latest/models.html#get-descendants-include-self-false

It seems you can use only this instead `Category.objects.select_related()...`
Probably it should make it a single query

Reply:
i'll subtask this investigation